lib/email_reply_parser.rb

```diff
@@ -132,8 +132,37 @@ def read(text)
 
   private
     EMPTY = "".freeze
+    # Detects `Sent from my new iPhone 7`
+    # or
+    # --
+    # or
+    # __
     SIGNATURE = '(?m)(--\s*$|__\s*$|\w-$)|(^(\w+\s*){1,3} ym morf tneS$)'
 
+
+    DETECT_QUOTE_REGEX = [
+      #En
+      /^On.*wrote:$/,
+      /^(From|Sent|To|Subject):.*$/,
+
+      # Fr
+      /^Le (jeu|lun|ven|dim|mar|sam|mer)\. \d{1,2} \w{3,5}\. .*$/,
+
+      # Nl
+      /^Op (ma|di|wo|do|vr|za|zo) \d{1,2} \w{3,5}\. .*<$/,
+
+      # It
+      /^Il giorno (lun|mar|mer|gio|ven|sab|dom) \d{1,2} \w{3,5}\ .*<$/,
+
+      # Line where the email is split into multiple lines
+      />:$/
```

Comment:
^ Added this to detect lines that are also considered quoted if they end in `>:`.